packages/core/src/config/parse.ts

```diff
@@ -1185,6 +1222,127 @@ const parseContractVariables = (
   return parsedConfigVariables
 }
 
+const parseArrayConstructorArg = (
+  input: ParamType,
+  constructorArgValue: UserConfigVariable
+): ParsedConfigVariable[] => {
+  if (!Array.isArray(constructorArgValue)) {
+    throw new InputError(
+      `Expected array for ${input.name} but got ${typeof constructorArgValue}`
+    )
+  }
+
+  if (input.arrayLength !== -1) {
+    if (constructorArgValue.length !== input.arrayLength) {
+      throw new InputError(
+        `Expected array of length ${input.arrayLength} for ${input.name} but got array of length ${constructorArgValue.length}`
+      )
+    }
+  }
+
+  const parsedValues: ParsedConfigVariable = []
+  for (const element of constructorArgValue) {
+    parsedValues.push(
+      parseAndValidateConstructorArg(input.arrayChildren, element)
+    )
+  }
+
+  return parsedValues
+}
+
+export const parseStructConstructorArg = (
+  paramType: ParamType,
+  constructorArgValue: UserConfigVariable
+) => {
```

Comment:
I kept this because I know it works and I'll need it when I add support for structs in a followup PR.